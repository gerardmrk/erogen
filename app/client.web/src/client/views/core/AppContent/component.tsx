import * as React from "react";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import { Switch } from "react-router";
import LoadingDisplay from "./LoadingDisplay";
import routeConfs from "@client/views/conf.routes";
import EnhancedRoute from "@client/views/components/EnhancedRoute";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class AppContent extends React.PureComponent<Props, State> {
  public render() {
    return (
      <main className={styles.main}>
        <React.Suspense fallback={<LoadingDisplay />}>
          <Switch>
            {routeConfs.map((r, i) => (
              <EnhancedRoute key={i} {...r} />
            ))}
          </Switch>
        </React.Suspense>
      </main>
    );
  }
}

export default AppContent;
