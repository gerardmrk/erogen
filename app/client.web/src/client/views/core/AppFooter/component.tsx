import * as React from "react";
import { Pane } from "evergreen-ui";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Container from "@client/views/components/Container";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class AppFooter extends React.PureComponent<Props, State> {
  public render() {
    return (
      <Pane className={styles.main} border={"muted"} background={"tint3"}>
        <Container className={styles.container}>{"AppFooter"}</Container>
      </Pane>
    );
  }
}

export default AppFooter;
