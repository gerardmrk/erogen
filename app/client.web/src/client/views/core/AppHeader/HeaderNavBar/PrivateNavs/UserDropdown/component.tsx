import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Icon from "@client/views/components/ui-elements/Icon";
import Image from "@client/views/components/ui-elements/Image";
import Dropdown from "@client/views/components/ui-modules/Dropdown";
import Placeholder from "@client/views/components/ui-elements/Placeholder";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class UserDropdown extends React.PureComponent<Props, State> {
  private onProfileNavClick = () => {
    if (this.props.profile) {
      this.props.history.push(`/user/${this.props.profile.username}`);
    }
  };

  private onDashboardNavClick = () => {
    this.props.history.push("/");
  };

  private onSettingsNavClick = () => {
    this.props.history.push("/settings");
  };

  private onLogoutActionClick = () => {
    this.props.logout();
  };

  public render() {
    const trigger =
      this.props.isLoadingUser || !this.props.profile ? (
        <Icon loading={true} circular={true} bordered={true} name={"spinner"} />
      ) : this.props.profile.displayPicUrl ? (
        <Image
          avatar={true}
          circular={true}
          bordered={true}
          src={this.props.profile.displayPicUrl}
        />
      ) : (
        <Icon circular={true} bordered={true} name={"user outline"} />
      );

    return (
      <Dropdown
        icon={null}
        trigger={trigger}
        direction={"left"}
        pointing={"top"}
      >
        <Dropdown.Menu direction={"left"}>
          <Dropdown.Header
            onClick={this.onProfileNavClick}
            className={styles.dropdownHeader}
          >
            {this.props.isLoadingUser || !this.props.profile ? (
              <Placeholder>
                <Placeholder.Header>
                  <Placeholder.Line length={"full"} />
                </Placeholder.Header>
              </Placeholder>
            ) : (
              <span>{this.props.profile.username}</span>
            )}
          </Dropdown.Header>

          <Dropdown.Divider />

          <Dropdown.Item onClick={this.onProfileNavClick}>
            {this.props.t("navs.private.profile")}
          </Dropdown.Item>

          <Dropdown.Item onClick={this.onDashboardNavClick}>
            {this.props.t("navs.private.dashboard")}
          </Dropdown.Item>

          <Dropdown.Divider />

          <Dropdown.Item onClick={this.onSettingsNavClick}>
            {this.props.t("navs.private.settings")}
          </Dropdown.Item>

          <Dropdown.Item onClick={this.onLogoutActionClick}>
            {this.props.t("navs.private.logout")}
          </Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>
    );
  }
}
