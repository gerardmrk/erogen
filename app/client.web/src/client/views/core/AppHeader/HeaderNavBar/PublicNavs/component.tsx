import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import NavLink from "@client/views/components/NavLink";
import { headerNavsPublic } from "@client/views/conf.navs";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class PublicNavs extends React.PureComponent<Props, State> {
  public render() {
    return (
      <nav className={styles.main}>
        <div className={styles.otherNavsGroup}>
          {headerNavsPublic.map((n, i) => (
            <div className={styles.navBox} key={i}>
              <NavLink path={n.to} label={n.label} />
            </div>
          ))}
        </div>

        <div className={styles.defaultNavsGroup}>
          <div className={styles.navBox}>
            <NavLink
              path={"/login"}
              label={this.props.t("navs.public.login")}
            />
          </div>

          <div className={styles.navBox}>{"/"}</div>

          <div className={styles.navBox}>
            <NavLink
              path={"/register"}
              label={this.props.t("navs.public.register")}
            />
          </div>
        </div>
      </nav>
    );
  }
}
