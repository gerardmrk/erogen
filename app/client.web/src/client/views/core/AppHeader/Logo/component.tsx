import * as React from "react";
import styles from "./component.styles.scss";
import { Link } from "react-router-dom";
import logo from "@client/logo.svg";
import { DEFAULT_PUBLIC_PATH } from "@client/views/conf.routes";
import { LocalProps } from ".";

export type Props = LocalProps & {};

export type State = {};

export class Logo extends React.PureComponent<Props, State> {
  public constructor(props) {
    super(props);
  }

  public render() {
    return (
      <div className={styles.main}>
        <Link to={DEFAULT_PUBLIC_PATH}>
          <img src={logo} alt={"brand logo"} />
        </Link>
      </div>
    );
  }
}
