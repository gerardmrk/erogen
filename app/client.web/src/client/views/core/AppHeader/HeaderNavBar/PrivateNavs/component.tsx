import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import UserDropdown from "./UserDropdown";
// import Menu from "@client/views/components/ui.collections/Menu";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class PrivateNavs extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <div className={styles.navs}>{"..."}</div>
        <div className={styles.userdropdown}>
          <UserDropdown />
        </div>
      </div>
    );
  }
}
