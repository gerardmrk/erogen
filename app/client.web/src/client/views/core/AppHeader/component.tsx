import * as React from "react";
import styles from "./styles.scss";
import { Pane } from "evergreen-ui/esm/layers";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Container from "@client/views/components/Container";
import Logo from "./Logo";
import HeaderNavBar from "./HeaderNavBar";

export type Props = LocalProps & StoreProps & DispatchProps;

export type State = {};

export class AppHeader extends React.PureComponent<Props, State> {
  public render() {
    return (
      <Pane
        borderColor={"x"}
        is={"header"}
        border={"muted"}
        background={"blueTint"}
      >
        <Container className={styles.container}>
          <div className={styles.logo}>
            <Logo />
          </div>
          <HeaderNavBar />
        </Container>
      </Pane>
    );
  }
}

export default AppHeader;
