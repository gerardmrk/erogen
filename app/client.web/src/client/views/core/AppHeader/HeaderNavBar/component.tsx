import * as React from "react";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import PrivateNavs from "./PrivateNavs";
import PublicNavs from "./PublicNavs";

export type Props = LocalProps & StoreProps & DispatchProps;
export type State = {};

export class HeaderNavBar extends React.Component<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        {this.props.isAuthenticated ? <PrivateNavs /> : <PublicNavs />}
      </div>
    );
  }
}

export default HeaderNavBar;
