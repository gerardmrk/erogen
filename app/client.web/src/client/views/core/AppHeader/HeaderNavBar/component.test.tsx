// Only enable snapshotting when component has stabilised and is not
// expected to undergo much development changes in the future.
const SNAPSHOT_ENABLED = true;

import * as React from "react";
import { ReactTestRenderer, ReactTestInstance } from "react-test-renderer"; // prettier-ignore
import { createRenderer } from "@client/views/_fixtures_/create-renderer";

import { HeaderNavBar } from "./component";

const Component = HeaderNavBar;

describe("<HeaderNavBar/>", () => {
  let renderer: ReactTestRenderer;
  let component: ReactTestInstance;

  beforeEach(() => {
    renderer = createRenderer(<Component isAuthenticated={true} />);
    component = renderer.getInstance() as ReactTestInstance;
  });

  if (SNAPSHOT_ENABLED) {
    test("snapshot", () => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  }

  it("renders ok", () => {
    expect(component).toBeDefined();
  });
});
