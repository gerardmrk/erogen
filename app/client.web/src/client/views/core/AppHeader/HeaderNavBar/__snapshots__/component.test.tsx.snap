// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderNavBar/> snapshot 1`] = `
<div
  className="main"
>
  <div
    className="main"
  >
    <div
      className="container"
    >
      PrivateNavs
    </div>
  </div>
</div>
`;
